src/main/org/firebirdsql/jdbc/InternalTransactionCoordinator.java

```diff
@@ -175,7 +175,7 @@ void handleConnectionClose() throws SQLException {
         coordinator.handleConnectionClose();
     }
 
-    private void setCoordinator(AbstractTransactionCoordinator coordinator) throws SQLException {
+    public void setCoordinator(AbstractTransactionCoordinator coordinator) throws SQLException {
```

Comment:
I currently can achieve it with:

setAutoCommit(true);
setAutoCommit(false);

Because with only setAutoCommit(false) it thinks it already is with autoCommit=false and does not change the coordinator.